### ISSUE TITLE
Weird panic when running on libcollection

### ISSUE THREAD

**Issue by mseri** [nrc/rustfmt#455]
I am sorry I cannot make a shorter example, I am not sure yet where the problem starts.
The panic happens running rustfmt on `src/libcollections/lib.rs`, current master branch.

The backtrace output (rustfm built with release target) is
```
RUST_BACKTRACE=1 rustfmt src/libcollections/lib.rs
thread '<main>' panicked at 'position 153500 does not resolve to a source location', ../src/libsyntax/codemap.rs:958
stack backtrace:
   1:        0x10c831c78 - sys::backtrace::tracing::imp::write::h13ce91411067cb99jRs
   2:        0x10c837e7f - panicking::log_panic::_<closure>::closure.38980
   3:        0x10c837922 - panicking::log_panic::h8e2e73ab7a77444fAFw
   4:        0x10c7fb3b6 - sys_common::unwind::begin_unwind_inner::h0d1483d9e4acd788VUr
   5:        0x10c7fbd5e - sys_common::unwind::begin_unwind_fmt::h6c37e0127f5942bb1Tr
   6:        0x10a0ecd8e - codemap::_<impl>::lookup_filemap_idx::h94a2fd6334b58366u9y
   7:        0x10a0ecb75 - codemap::_<impl>::bytepos_to_file_charpos::hc179b8243def8aaeZ5y
   8:        0x10a0ec8c1 - codemap::_<impl>::lookup_char_pos::h96ab3a8681081916EOy
   9:        0x10a0a0ac0 - codemap::_<impl>::lookup_char_pos_adj::had948ba6acd8b9d58Vy
  10:        0x10a0ece0b - codemap::_<impl>::span_to_string::h064a5b7faf3e7109BWy
  11:        0x10a13398b - diagnostic::_<impl>::emit_::h3f6f5a1881fa39a8JyA
  12:        0x10a131488 - diagnostic::_<impl>::emit::h0013c90a893f1b28scB
  13:        0x10a1524aa - parse::parser::_<impl>::check_strict_keywords::heb2f47d4a68abab32OE
  14:        0x10a15208f - parse::parser::_<impl>::parse_ident::ha9cc1e6907cc1257BHE
  15:        0x10a1699cb - parse::parser::_<impl>::parse_path_segments_with_colons::ha8e40127a02d59ad8oG
  16:        0x10a156ef4 - parse::parser::_<impl>::parse_path::ha204331161a1ad29EiG
  17:        0x10a16f20b - parse::parser::_<impl>::parse_bottom_expr::h09b8c4a7c413ea9eFGG
  18:        0x10a172f5d - parse::parser::_<impl>::parse_dot_or_call_expr::h7ae0d1d4ba381d28R9G
  19:        0x10a179354 - parse::parser::_<impl>::parse_prefix_expr::hf3586ede11ec473b6CH
  20:        0x10a17995d - parse::parser::_<impl>::parse_binops::h8a1624a747db152bMJH
  21:        0x10a17a31e - parse::parser::_<impl>::parse_assign_expr::h3dd685951257dd6bEPH
  22:        0x10a1795ec - parse::parser::_<impl>::parse_expr_res::hc6b28b30bb2448b9NgI
  23:        0x10a16d19b - parse::parser::_<impl>::parse_bottom_expr::h09b8c4a7c413ea9eFGG
  24:        0x10a172f5d - parse::parser::_<impl>::parse_dot_or_call_expr::h7ae0d1d4ba381d28R9G
  25:        0x10a179354 - parse::parser::_<impl>::parse_prefix_expr::hf3586ede11ec473b6CH
  26:        0x10a17995d - parse::parser::_<impl>::parse_binops::h8a1624a747db152bMJH
  27:        0x10a17a31e - parse::parser::_<impl>::parse_assign_expr::h3dd685951257dd6bEPH
  28:        0x10a145fec - parse::parser::_<impl>::parse_expr::h8220dc7185f813a4IoE
  29:        0x1088da377 - sys_common::unwind::try::try_fn::try_fn::h8329852188099625790
  30:        0x10c82f828 - __rust_try
  31:        0x10c8244be - sys_common::unwind::try::inner_try::h9c69b1952218bc71tRr
  32:        0x10888992d - macros::rewrite_macro::h96d75bff23db60f7yfk
  33:        0x108881670 - expr::_<impl>::rewrite::haf0d3076bb637c78bLg
  34:        0x10887fd89 - visitor::_<impl>::visit_expr::h8fb4721c80a57ef3tnd
  35:        0x108887003 - visitor::_<impl>::visit_block::hdc633d9d07367369Ird
  36:        0x10888764d - visitor::_<impl>::visit_fn::h9b650b9e9683c73elvd
  37:        0x108889518 - visit::walk_impl_item::walk_impl_item::h9934228292691353229
  38:        0x108888855 - visit::walk_item::walk_item::h10320707368896113990
  39:        0x108883ff9 - visitor::_<impl>::visit_item::hb37bfc20f0d924f8wxd
  40:        0x1088df50f - _<impl>::build_controller::_<closure>::closure.28495
  41:        0x1089b4846 - driver::compile_input::hcfd2fd1b866a737f7ba
  42:        0x108b0c700 - run_compiler::h08f9725ff9bbf279Rpc
  43:        0x1088e112e - format::hef17712dbd87bb96IHk
  44:        0x1088e1b92 - run::hd0d52b0fdb26b5e7xIk
  45:        0x108849a8f - main::h5ca3068261bd1bc80da
  46:        0x10c8376a2 - sys_common::unwind::try::try_fn::h7739041069973150093
  47:        0x10c82f828 - __rust_try
  48:        0x10c837549 - rt::lang_start::h3d1e8cf112639bfcICw
```

Recompiling rustfmt with the debug target I get:
```
thread '<main>' panicked at 'position 153500 does not resolve to a source location', ../src/libsyntax/codemap.rs:958
stack backtrace:
   1:        0x1106edc78 - sys::backtrace::tracing::imp::write::h13ce91411067cb99jRs
   2:        0x1106f3e7f - panicking::log_panic::_<closure>::closure.38980
   3:        0x1106f3922 - panicking::log_panic::h8e2e73ab7a77444fAFw
   4:        0x1106b73b6 - sys_common::unwind::begin_unwind_inner::h0d1483d9e4acd788VUr
   5:        0x1106b7d5e - sys_common::unwind::begin_unwind_fmt::h6c37e0127f5942bb1Tr
   6:        0x10dfadd8e - codemap::_<impl>::lookup_filemap_idx::h94a2fd6334b58366u9y
   7:        0x10dfadb75 - codemap::_<impl>::bytepos_to_file_charpos::hc179b8243def8aaeZ5y
   8:        0x10dfad8c1 - codemap::_<impl>::lookup_char_pos::h96ab3a8681081916EOy
   9:        0x10df61ac0 - codemap::_<impl>::lookup_char_pos_adj::had948ba6acd8b9d58Vy
  10:        0x10dfade0b - codemap::_<impl>::span_to_string::h064a5b7faf3e7109BWy
  11:        0x10dff498b - diagnostic::_<impl>::emit_::h3f6f5a1881fa39a8JyA
  12:        0x10dff2488 - diagnostic::_<impl>::emit::h0013c90a893f1b28scB
  13:        0x10e0134aa - parse::parser::_<impl>::check_strict_keywords::heb2f47d4a68abab32OE
  14:        0x10e01308f - parse::parser::_<impl>::parse_ident::ha9cc1e6907cc1257BHE
  15:        0x10e02a9cb - parse::parser::_<impl>::parse_path_segments_with_colons::ha8e40127a02d59ad8oG
  16:        0x10e017ef4 - parse::parser::_<impl>::parse_path::ha204331161a1ad29EiG
  17:        0x10e03020b - parse::parser::_<impl>::parse_bottom_expr::h09b8c4a7c413ea9eFGG
  18:        0x10e033f5d - parse::parser::_<impl>::parse_dot_or_call_expr::h7ae0d1d4ba381d28R9G
  19:        0x10e03a354 - parse::parser::_<impl>::parse_prefix_expr::hf3586ede11ec473b6CH
  20:        0x10e03a95d - parse::parser::_<impl>::parse_binops::h8a1624a747db152bMJH
  21:        0x10e03b31e - parse::parser::_<impl>::parse_assign_expr::h3dd685951257dd6bEPH
  22:        0x10e03a5ec - parse::parser::_<impl>::parse_expr_res::hc6b28b30bb2448b9NgI
  23:        0x10e02e19b - parse::parser::_<impl>::parse_bottom_expr::h09b8c4a7c413ea9eFGG
  24:        0x10e033f5d - parse::parser::_<impl>::parse_dot_or_call_expr::h7ae0d1d4ba381d28R9G
  25:        0x10e03a354 - parse::parser::_<impl>::parse_prefix_expr::hf3586ede11ec473b6CH
  26:        0x10e03a95d - parse::parser::_<impl>::parse_binops::h8a1624a747db152bMJH
  27:        0x10e03b31e - parse::parser::_<impl>::parse_assign_expr::h3dd685951257dd6bEPH
  28:        0x10e006fec - parse::parser::_<impl>::parse_expr::h8220dc7185f813a4IoE
  29:        0x10c334e81 - macros::rewrite_macro::_<closure>::closure.28696
  30:        0x10c33463d - std::thread::catch_panic::_<closure>::closure.28679
  31:        0x10c3345d9 - sys_common::unwind::try::try_fn::try_fn::h9126153315004067307
  32:        0x1106eb828 - __rust_try
  33:        0x1106e04be - sys_common::unwind::try::inner_try::h9c69b1952218bc71tRr
  34:        0x10c334547 - sys_common::unwind::try::try::h12336003261990977086
  35:        0x10c33428c - thread::catch_panic::catch_panic::h8917902472524598890
  36:        0x10c2573b4 - macros::rewrite_macro::h7d356d23fc117237yfk
  37:        0x10c243aa4 - expr::_<impl>::rewrite::h053d57507e446145bLg
  38:        0x10c24107f - visitor::_<impl>::visit_expr::he3ea3720598a0ea2tnd
  39:        0x10c24ca1f - visitor::_<impl>::visit_block::hdb18d82a604588f7Ird
  40:        0x10c24dc6c - visitor::_<impl>::visit_fn::hf05789cfdc746fc1lvd
  41:        0x10c256dea - visit::walk_impl_item::walk_impl_item::h14760420592400086900
  42:        0x10c2542da - visitor::_<impl>::visit_impl_item::h05c4a1693b0f6e170Bd
  43:        0x10c24f92c - visit::walk_item::walk_item::h6769836250362607554
  44:        0x10c24bddd - visitor::_<impl>::visit_item::hac187297a600ec37wxd
  45:        0x10c25086e - visit::walk_mod::walk_mod::h9347762371244949737
  46:        0x10c25cf27 - visitor::_<impl>::format_separate_mod::h32daf9d422cd88fbsJd
  47:        0x10c356c3f - fmt_ast::ha83a58bd9f324f7dPyk
  48:        0x10c365fcc - _<impl>::build_controller::_<closure>::closure.29309
  49:        0x10c87a846 - driver::compile_input::hcfd2fd1b866a737f7ba
  50:        0x10c9d2700 - run_compiler::h08f9725ff9bbf279Rpc
  51:        0x10c366855 - format::h21767ac575acc800IHk
  52:        0x10c367aa7 - run::hd7727a6bc0d60011xIk
  53:        0x10c1e478c - execute::ha02357fc43f76710Hca
  54:        0x10c1e75d0 - main::h5ca3068261bd1bc80da
  55:        0x1106f36a2 - sys_common::unwind::try::try_fn::h7739041069973150093
  56:        0x1106eb828 - __rust_try
  57:        0x1106f3549 - rt::lang_start::h3d1e8cf112639bfcICw
  58:        0x10c1e8d09 - main
```

I noticed that there is no error if I run rustfmt on single files, e.g. with
```
for i in **/*.rs
do
  echo $i
  rustfmt $i
done
```

**Comment by marcusklaas** [comment]
This is just a macro use that the parser fails to parse. The panic is contained in its own thread and rustfmt should continue. Unfortunately there's no way to prevent the panic from writing to stdout/ stderr (yet)...

**Comment by nrc** [comment]
We could redirect stderr/out while we call out to the parser. Not actually sure how to do that, but I know the compiler test infra does it.

**Comment by marcusklaas** [comment]
Nice, that should be an interesting exercise!

**Comment by TimNN** [comment]
This may have been fixed by #543 - the panic message was the same as #538 even though the backtrace was different.

Also, I'm a little confused by @marcusklaas's statement that the panic is contained in its own thread because the backtrace starts at `main` and is on thread `<main>`.

**Comment by marcusklaas** [comment]
The panic used to be contained in a `catch_panic` call. I think we don't actually started a separate thread for it though, so that was inaccurate of me. This was recently changed when we moved to rust stable.

Anyway, this should be fixed now.
